### PR TITLE
init_desktop_runner: ensure we have plasma-krunner suggestions

### PR DESCRIPTION
**Pull request by ggardet**
- Verification runs: 
  * https://openqa.opensuse.org/tests/932385#step/desktop_runner/10
  * https://openqa.opensuse.org/tests/932413

### CODE REVIEW

File: lib/susedistribution.pm [hunk 1]
```diff
@@ -154,25 +154,33 @@ sub init_desktop_runner {
...
+                type_string_very_slow substr $program, 2;
+            } else {
+                type_string_slow $program;
+            }
+        } else {
+            type_string $program;
         }
-        else {
-            type_string_slow $program;
+        # Make sure we have plasma suggestions as it may take time, especially on boot or under load. Otherwise, try again
```
Comment (foursixnine) [inline lib/susedistribution.pm]: That's clever

**Comment by ggardet** [comment]
> Looks fine from my point of view, but could you schedule ~20-30 runs? 

* 20 kde tests on Leap 15.1 arm: https://openqa.opensuse.org/tests/overview?distri=opensuse&groupid=38&build=pr7442_verification_run&version=15.1
* 20 kde tests on tw x86_64: https://openqa.opensuse.org/tests/overview?version=Tumbleweed&build=pr7442_verification_run&distri=opensuse&groupid=38

**Comment by foursixnine** [comment]
Let's merge for now... We know what happened if there's a problem in this area :)

**Comment by StefanBruens** [comment]
This breaks all the KDE tests on TW, the code always hits the `!check_screen('desktop-runner-plasma-suggestions')` case in the loop.

check_screen with a timeout of 0 (implicit here) seems to not run the check at all, see e.g. log from
https://openqa.opensuse.org/tests/941393

```
[2019-05-24T11:57:58.995 CEST] [debug] <<< testapi::type_string(string='xterm', max_interval=13, wait_screen_changes=0, wait_still_screen=0, timeout=30, similarity_level=47)
[2019-05-24T11:57:58.995 CEST] [debug] /var/lib/openqa/cache/openqa1-opensuse/tests/opensuse/tests/x11/chromium.pm:22 called testapi::ensure_installed
[2019-05-24T11:57:58.995 CEST] [debug] <<< testapi::wait_screen_change(timeout=10, similarity_level=50)
[2019-05-24T11:58:00.515 CEST] [debug] waiting for screen change: 0 14.5498245783459
[2019-05-24T11:58:00.515 CEST] [debug] >>> testapi::wait_screen_change: screen change seen at 0
[2019-05-24T11:58:00.516 CEST] [debug] /var/lib/openqa/cache/openqa1-opensuse/tests/opensuse/tests/x11/chromium.pm:22 called testapi::ensure_installed
[2019-05-24T11:58:00.516 CEST] [debug] <<< testapi::check_screen(mustmatch='desktop-runner-plasma-suggestions', timeout=0)
[2019-05-24T11:58:00.738 CEST] [debug] >>> testapi::_check_backend_response: match=desktop-runner-plasma-suggestions timed out after 0 (check_screen)
[2019-05-24T11:58:00.779 CEST] [debug] /var/lib/openqa/cache/openqa1-opensuse/tests/opensuse/tests/x11/chromium.pm:22 called testapi::ensure_installed
[2019-05-24T11:58:00.779 CEST] [debug] <<< testapi::send_key(key='esc', do_wait=0)
```
Though this is IMHO a bug in check_screen, can you set the timeout to some larger value to test if this fixes the problem?

**Comment by DimStar77** [comment]
This kind of issue could be related: https://openqa.opensuse.org/tests/941399 ?

**Comment by StefanBruens** [comment]
A timeout of 10 times 30 seconds seems quite excessive for me - the tests in TW currently fail due to a mismatching needle (sharpened configure icon), and each test now takes 5 minutes to complete.

**Comment by DimStar77** [comment]
I see - and @nilxam already created a new, matching needle: https://github.com/os-autoinst/os-autoinst-needles-opensuse/commit/129755adb3287bf906dd21ec6feb2a877744b3ba

**Comment by StefanBruens** [comment]
(My previous comment was actually wrong, the check *is* run, but nothing is logged. Did a PR to avoid the silent log, os-autoinst/os-autoinst#1157)

**Comment by okurz** [comment]
@foursixnine @ggardet please take into consideration what had been done already in the (recent) past trying to fix the situation. I tend to reverting this change completely due to https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/7679#pullrequestreview-249777560 . Running 20 jobs just does not cut it , see https://progress.opensuse.org/issues/35589 as well with many more runs trying to find out where is the sweet spot between stable and wasting a lot of time.

**Pull request merged**